Data access comparison with AQUA scritps

mnurisso opened the pull request:
A more complete description will follow (still WIP)

Just a script and custom catalog skeleton so far.

mnurisso committed:
script skeleton

mnurisso committed:
catalog skeleton

mnurisso committed:
region option

mnurisso committed:
loop on the different exp generated to test the data access

mnurisso committed:
script to install the custom aqua catalog

mnurisso committed:
aqua installation script

mnurisso committed:
final version of the catalog

mnurisso committed:
time benchmark added to the script

mnurisso committed:
job adapted to complete LRA 1 2d variable for the three available experiments

mnurisso committed:
little test sources notebook

mnurisso committed:
introduction of a minimal .gitignore file

mnurisso committed:
nproc customizable option

mnurisso committed:
test for various worker/memory setup

mnurisso committed:
08 and 16 GB scripts

mnurisso committed:
better scaling plot visualization

mnurisso committed:
removed old notebook

mnurisso committed:
moved LRA test in a specific subfolder

mnurisso committed:
slurm array jobs

mnurisso committed:
script to produce 1 month of r100 data for benchmark

mnurisso committed:
results

mnurisso committed:
added engine argument to test polytope data access

mnurisso committed:
results and plot script

mnurisso committed:
better legend

mnurisso committed:
updated plot

mnurisso committed:
working notebook for smmregrid integrated grids

mnurisso committed:
attempt of a smmregrid benchmark script

mnurisso committed:
starting point notebook

koldunovn commented:
@mnurisso Tell me when to merge - we don't have to make it a gigantic PR, can add things on the way :)

mnurisso committed:
added grib files to .gitignore

mnurisso committed:
soft scaling data

mnurisso committed:
further results

mnurisso commented:
> @mnurisso Tell me when to merge - we don't have to make it a gigantic PR, can add things on the way :)

Hi @koldunovn I just worked on this branch on my own since there was no dependency on other parts of the repo so far.
I'm trying to clean a bit the scripts and this can be merged.

mnurisso committed:
updated readme file

mnurisso committed:
moved install script

mnurisso committed:
removed unused LRA variations

mnurisso committed:
update for possible chunk in filename

mnurisso committed:
new data for figure

mnurisso committed:
deleted old visuals

mnurisso committed:
removed wrong result

mnurisso commented:
The PR can be merged and I can keep work on other details. What I've done so far is to test AQUA under different configurations, in particular with:

- strong scaling: fixed memory, increasing workers (1 to 64), from hourly healpix9 to monthly 1 deg compared against a perfect scaling (plot.ipynb notebook)
- weak scaling:  fixed memory, increasing number of workers and increasing number of months to process simultaneously. A bit of degradation but weak scaling stays quite decent.
- polytope test: tested 1 month as in the first case with polytope (GSV and AQUA using internally polytope credentials). I see a factor 4 slowdown and mostly important way more attention to chunking based on the number of workers is needed (I used weekly chunking instead of daily in order to minimize the number of downloads with polytope). High fluctuations if the request is queued. Tests suggests that polytope is very slow but still is possible to process high-res data.

An LRA folder is ready to test more extensive tasks like 5 years of LRA or other data reduction.
A folder to compare smmregrid against CDO is set, just a skeleton so far, will be completed in another PR.

mnurisso committed:
larger fonts
